src/propmachinery.jl

```diff
@@ -1,32 +1,95 @@
-import Base: start, next, done
-
 abstract QuPropagatorMethod
 
-immutable QuPropagator{QPM<:QuPropagatorMethod}
-    hamiltonian
-    init_state::QuVector
+abstract QuantumEquation
```

Comment:
I think we should stick to the naming convention we used so far and call this `QuEquation`. Maybe the other types also should get a Qu-prefix?